### PR TITLE
Add flag in the core/ui data store to indicate when we are activating…

### PR DESCRIPTION
**Pull request by asvinb**
… a module.

## Summary

Addresses issue #4267 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @asvinb. Looks like I missed it during reviewing IB but we need to implement it slightly different. Check what AC says:

> A progress bar should replace the "Set up Analytics" link/button in the bottom of the CTA widget, so there is no CTA button/link in the widget and the user gets a visual indication that something is happening before/during the redirect

It means that we need to display the progress bar instead of the setup button in the ActivateModuleCTA component itself and we don't need the progress bar in the Overview component of the SearchFunnelWidget widget.

Could you please update your PR to match requirements from AC?

**Comment by asvinb** [comment]
@eugene-manuilov Thanks for catching that. I have updated the PR to reflect that in the following [commit](https://github.com/google/site-kit-wp/pull/4324/commits/2676d3026e5ceb4f2e1bb3753363a61c5227aec4). I have added a new `inProgress` prop to the `CTA` component since it's what being used. My main worry right now however is how that change can affect the whole site, meaning whenever we have a CTA to activate a module, the progress bar will be displayed. I'am perfectly fine with that but curious whether the impact of it.
cc @tofumatt 

Let me know what you think.

File: assets/js/components/notifications/CTA.js [hunk 1]
```diff
@@ -51,7 +53,14 @@ const CTA = ( {
 				{ description }
 			</div>
 		) }
-		{ ctaLabel && (
+
+		{ inProgress && (
+			<div className="googlesitekit-cta__progress">
+				<ProgressBar />
```
Comment (eugene-manuilov) [inline assets/js/components/notifications/CTA.js:59]: Let's add the `small` attribute to the progress bar to make it look nicer: 
```suggestion
				<ProgressBar small />
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Awesome! Thanks, @asvinb. This looks almost ideal to me. Added one suggestion after which we can approve this.

File: assets/js/components/notifications/CTA.js (hunk 1 above)
Comment (asvinb) [reply assets/js/components/notifications/CTA.js]: 👍

**Review by eugene-manuilov — APPROVED** [review]
Thanks, @asvinb. Looks good to me.

**Comment by felixarntz** [comment]
@asvinb What's the state of this PR here? Without having looked to closely, I'm slightly confused because this is still open, but the issue #4267 is successfully closed including QA 🤔 

Was there an alternative PR that fixed this?

**Comment by asvinb** [comment]
@felixarntz Apologies for the late reply, it's only now i saw your comment. I'am puzzled as well. Most likely another PR fixed the issue but I'll dig into it and find out what exactly happened.

**Comment by aaemnnosttv** [comment]
Closing as the related issue has already been addressed.

**Pull request closed without merging**